### DISCUSSION TITLE
How to apply plugin through Gradle script plugin

### DISCUSSION THREAD

**Discussion by vladimirfx** [Splitties/refreshVersions#317] [Q&A] [answered]
First thank you very mach for such useful plugin! We use it from it early days (buildSrc) widely in our company projects from Android to microservices.

Because of wide adoption we need a way to apply your plugin for nearly all out projects in consistent way. For such cases we use common Gradle script plugin project through git subrepo in our projects. So I've tried to apply refreshVersions this way using bootstrap for buildSrc function. Script compiled but build is crashed.

Does such use case supported or planned?

https://docs.gradle.org/current/userguide/plugins.html#sec:script_plugins

**Comment by LouisCAD** [comment, marked as answer]
Hello, sorry for replying so late.
refreshVersions can only be configured in `settings.gradle[.kts]` files. You have one for the root project, and you can have one for the root project of your `buildSrc` as well. It's then applied to all subProjects.
